torch/compiler/__init__.py

```diff
@@ -216,18 +217,69 @@ def assume_constant_result(fn):
 
 def disable(fn=None, recursive=True):
     """
-    This function provides both a decorator and a context manager to disable compilation on a function
+    This function provides a decorator manager to disable compilation on a function
     It also provides the option of recursively disabling called functions
 
+    .. note::
+        Interaction between `compile`, `disable`, and `enable`
+
+        `compile` is is a "marker" that Dynamo should attempt to compile the function
+        and its nested calls.
+
+        `disable` is higher-priority - it signifies that a function (and its nested
+        calls in the case recursive=True) should not be compiled.
```

Comment:
I think that we should explicitly describe the model where there is a flag (is compile enabled/disabled) and then describe these API functions in terms of the flag.